ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -62,6 +62,15 @@ const (
 	InteractiveMode JobSubmissionMode = "InteractiveMode" // Don't submit job in KubeRay. Instead, wait for user to submit job and provide the job submission ID.
 )
 
+type DeletionPolicy string
+
+const (
+	DeleteClusterDeletionPolicy DeletionPolicy = "DeleteCluster"
+	DeleteWorkersDeletionPolicy DeletionPolicy = "DeleteWorkers"
+	DeleteSelfDeletionPolicy    DeletionPolicy = "DeleteSelf"
+	DeleteNoneDeletionPolicy    DeletionPolicy = "DeleteNone"
```

Comment:
In some cases, users may want to have different delete policies for Ray jobs that succeeded and Ray jobs that failed. For example, `DeleteCluster` if job succeeded and `DeleteWorkers` if job failed.

Do we plan to support these cases? We don't need to include these policies in this PR, but I am wondering what the APIs would look like.

Reply:
I think the API is flexible enough to support these in the future. `DeleteClusterOnSuccess` or `DeleteWorkersOnFailure`, etc